### PR TITLE
[ci] upgrade GitHub Actions third-party actions to newest versions

### PR DESCRIPTION
**Pull request by jameslamb**
I've been noticing a bunch of these warnings in GitHub Actions jobs:

> The `add-path` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

> The `set-env` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

For example, you can see these annotations on https://github.com/microsoft/LightGBM/actions/runs/343069981.

To be sure that these changes from GitHub don't disrupt our CI, this PR updates the GitHub Actions workflows in this project to use the newest versions of third-party actions. You can see, specifically, that updates to deal with this deprecation were made in the most recent release of `actions/checkout`: https://github.com/actions/checkout/releases/tag/v2.3.4

It also fixes a mistake I made setting them up....we should be using https://github.com/conda-incubator/setup-miniconda, not this fork of it: https://github.com/goanpeca/setup-miniconda.

### CODE REVIEW

**Review by StrikerRUS — CHANGES_REQUESTED** [review]
Please update this line as well.
https://github.com/microsoft/LightGBM/blob/12e220b79d459396ce74f60f260786ad67a8fbb9/.github/workflows/cuda.yml#L41

**Comment by jameslamb** [comment]
> Please update this line as well.
> https://github.com/microsoft/LightGBM/blob/12e220b79d459396ce74f60f260786ad67a8fbb9/.github/workflows/cuda.yml#L41

ah missed that one!

File: .github/workflows/r_valgrind.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
     container: wch1/r-debug
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v1
+        uses: actions/checkout@v2.3.4
```
Comment (jameslamb) [inline .github/workflows/r_valgrind.yml:16]: /gha run r-valgrind

File: .github/workflows/r_package.yml [hunk 2]
```diff
@@ -165,8 +165,13 @@ jobs:
     runs-on: ubuntu-latest
     container: rhub/rocker-gcc-san
     steps:
+      - name: Install Git before checkout
+        shell: bash
+        run: |
+          apt-get update
+          apt-get install -y git
```
Comment (StrikerRUS) [inline .github/workflows/r_package.yml:172]:
```suggestion
          apt-get install --no-install-recommends -y git
```

**Review by StrikerRUS — APPROVED** [review]
Thanks! Just one minor comment below.

File: .github/workflows/r_valgrind.yml (hunk 1 above)
Comment (jameslamb) [reply .github/workflows/r_valgrind.yml:16]: looks like valgrind is happy with this change. I did check before that the image we use for valgrind tests had a recent version of `git` in it, and it does.

https://github.com/microsoft/LightGBM/runs/1349685476?check_suite_focus=true

**Review by guolinke — APPROVED** [review]

**Pull request merged**